lib/ansible/modules/system/pkt_steering.py

```diff
@@ -0,0 +1,383 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# pkt_steering is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: pkt_steering
+short_description: Control active and configured kernel packet steering settings
+description:
+     - This module controls active and configured kernel packet steering
+       settings in a convenient way. (ref.
+       U(https://www.kernel.org/doc/Documentation/networking/scaling.txt))
+version_added: "2.4"
+options:
+    tech:
+        description:
+         - Packet Steering technology to configure.
+        required: true
+        choices: ['rfs', 'rps', 'xps']
+    state:
+        description:
+         - State of the specific packet steering technology. The state also
+           decides about the steering resources distribution. For xps/rps
+           C(absent), C(balanced), C(specific) are allowed. C(balanced)
+           purposes to bring about a benevolent distribution with respecting
+           also numa arches intrinsics like a split node cache hierarchy.  Rfs
+           only discerns C(present) and C(absent).
+        required: true
+        choices : ['present', 'absent', 'balanced', 'specific']
+    iface:
+        description:
+         - Kernel net interface steering to configure on. E.g. eth0, or enp2s0.
+        required : true
+        type: string
+    distribution:
+        description: >
+          -  Path to json file holding the individually thought up packet
+             steering queues to cpu association. Available for C(tech=xps) or
+             C(tech=rps) - it's useful when something else than balanced
+             steering is required.
+
+             Basic example:
+             {
+                 "distr": [
+                 { "0": [1, [4, 5]] },
+                 { "1": [1, [2, 3]] },
+                   [...]
+                 ]
+             }
+             Would spread queue 0 to cpus 1,4-5 and so on. Required when
+             C(state=specific).
+        required : false
+        default: None
+    tab_size:
+        description:
+          -  Kernel RFS flow dissection table size for rfs.
+             Optional when C(tech=rfs).
+        required : false
+        type: int
+        default: 32768
+
+author:
+    - Matthias Tafelmeier (@cherusk)
+'''
+
+RETURN = '''
+'''
+
+EXAMPLES = '''
```

Comment:
Examples must be written with a `- name:`, and using multi line `key: value`.

Please see https://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#examples-block for details